File: 2023/day_01/1.py
#!/usr/bin/env python3

with open('./input.txt') as file:
    lines = file.readlines()
    sum = 0
    for line in lines: 
        filtered = []
        for char in line:
            if char in "0123456789":
                filtered.append(int(char))
        sum += filtered[0]*10 + filtered[-1]
    print(sum)

File: 2024/day_01/1.jl
#!/usr/bin/env julia 

lines = []

open("./input.txt") do file
  for line in eachline(file)
    push!(lines,line)
  end
end

cols = map(x -> sort(x),
  eachrow(
    reshape(
      vcat(map(
        line -> map(n -> parse(Int, n), split(line)), 
        lines)...),
      (2,:))))

println(sum([abs(pair[1]-pair[2]) for pair in zip(cols[1], cols[2])]))
